### PR TITLE
UnknownCommandEvent support Adventure Component

### PR DESCRIPTION
**Pull request by LemonCaramel**
Current UnknownCommandEvent does not full support Adventure Components.
This PR helps fully support Adventure Components.

### CODE REVIEW

**Review by JRoy — APPROVED** [review]
good catch!

File: Spigot-Server-Patches/0151-Add-UnknownCommandEvent.patch [hunk 1]
```diff
@@ -5,35 +5,27 @@ Subject: [PATCH] Add UnknownCommandEvent
...
 diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
-index 5d357b0f84b5242066dcce203752a0f46e9a249c..4f9c42a4b0256f181263bf5e0492714a01fbec38 100644
+index 268d56859a0de3b2ca39155f882991b5bf3aa5e0..54b68bad12f22d793e02f4583d7ab12ba3c54dc6 100644
 --- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
 +++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
-@@ -104,6 +104,7 @@ import net.minecraft.world.level.storage.WorldNBTStorage;
- import net.minecraft.world.level.storage.loot.LootTableRegistry;
- import net.minecraft.world.phys.Vec3D;
- import org.apache.commons.lang.Validate;
-+import org.apache.commons.lang3.StringUtils;
```
Comment (kashike) [inline Spigot-Server-Patches/0151-Add-UnknownCommandEvent.patch:15]: Don't remove imports.

File: Spigot-Server-Patches/0151-Add-UnknownCommandEvent.patch [hunk 2]
```diff
@@ -5,35 +5,27 @@ Subject: [PATCH] Add UnknownCommandEvent
...
- import org.bukkit.BanList;
- import org.bukkit.Bukkit;
- import org.bukkit.ChatColor;
-@@ -177,6 +178,7 @@ import org.bukkit.craftbukkit.util.Versioning;
+@@ -177,6 +177,7 @@ import org.bukkit.craftbukkit.util.Versioning;
  import org.bukkit.craftbukkit.util.permissions.CraftDefaultPermissions;
  import org.bukkit.entity.Entity;
  import org.bukkit.entity.Player;
 +import org.bukkit.event.command.UnknownCommandEvent; // Paper
```
Comment (kashike) [inline Spigot-Server-Patches/0151-Add-UnknownCommandEvent.patch]: Fully-qualify instead.

**Review by kashike — CHANGES_REQUESTED** [review]

File: Spigot-Server-Patches/0151-Add-UnknownCommandEvent.patch (hunk 1 above)
Comment (LemonCaramel) [reply Spigot-Server-Patches/0151-Add-UnknownCommandEvent.patch:15]: I removed it because it was an unused import.
rolled back again!

Comment (jpenilla) [reply Spigot-Server-Patches/0151-Add-UnknownCommandEvent.patch:15]: Hmm... it looks like this import actually should be removed. It's added by the patch which is being modified here, and is no longer in use due to changes in this PR.

**Review by jpenilla — APPROVED** [review]
LGTM, sorry about the confusion with imports, we can resolve that on merge

**Review by kashike — APPROVED** [review]
for @jpenilla

**Pull request merged**